iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -497,8 +497,8 @@ LogicalResult convertInterfaceStoreTensorOp(
   b.setInsertionPoint(storeOp);
   Value storeTo = bvm.lookup(storeOp.target());
   // If the value already has a mapping, it should already have been updated in
-  // place by the converted producer.
-  if (storeTo) {
+  // place by the converted producer, as long as the producer is a Linalg op.
+  if (storeTo && storeOp.value().getDefiningOp<linalg::LinalgOp>()) {
```

Comment:
This is sort of a hack though (granted the whole pass is sort of hacky, but bufferization is hard). Maybe worth a more principled solution here.
Here is the knowns
1) storeOp.target() is mapped to a `hal.interface.binding.subspan`.
2) storeOp.value() is either unmapped, or already mapped to a subview. If latter, the subview base should be the same as the buffer `storeOp.target` is mapped to.

In general case we need some alias analysis mechanism, but I think that should be good enough for a while?

Reply:
Right. That's better than this specific check. Updated.